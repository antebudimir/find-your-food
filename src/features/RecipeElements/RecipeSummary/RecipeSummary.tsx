import {
	RecipeSummaryContainer,
	SummaryImage,
	SummaryTitle,
	SummaryLink,
} from './RecipeSummary.style';

interface RecipeSummaryProps {
	label?: string;
	image?: string;
	recipeId?: string;
}

const RecipeSummary = ({
	label = '',
	image = '',
	recipeId = '',
}: RecipeSummaryProps) => {
	return (
		<RecipeSummaryContainer>
			<SummaryTitle>{label ? label : 'Title not available'}</SummaryTitle>

			<SummaryImage
				imageSource={image ? image : 'Photo not available'}
				imageAlt={label ? label : 'Title not available'}
			/>

			<SummaryLink to={`recipe/${recipeId}`}>View Recipe Details</SummaryLink>
		</RecipeSummaryContainer>
	);
};

export default RecipeSummary;
