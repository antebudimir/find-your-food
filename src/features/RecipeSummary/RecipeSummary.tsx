import RecipeLine from 'features/RecipeLine/RecipeLine';
import RecipeList from 'features/RecipeList/RecipeList';
import {
	RecipeSummaryContainer,
	SummaryImage,
	SummaryTitle,
} from './RecipeSummary.style';
import { v4 as uuidv4 } from 'uuid';

interface RecipeSummaryProps {
	label?: string;
	image?: string;
	mealType?: string;
	ingredientLines?: string[];
	totalTime?: string;
}

const RecipeSummary = ({
	label,
	image,
	mealType,
	ingredientLines,
	totalTime,
}: RecipeSummaryProps) => {
	// create a list
	function createList(items: string[]) {
		const list = items.map((item) => {
			return <li key={uuidv4()}>{item}</li>;
		});
		return list;
	}

	return (
		<RecipeSummaryContainer>
			<SummaryTitle>{label ? label : 'Title not available'}</SummaryTitle>

			<SummaryImage
				imageSource={image ? image : 'Photo not available'}
				imageAlt={label ? label : 'Image description not available'}
			/>

			<RecipeLine
				label="Meal type"
				recipeDetail={mealType && mealType.length > 0 ? mealType : 'No info'}
			/>
			<RecipeLine
				label="Meal type"
				recipeDetail={mealType && mealType.length > 0 ? mealType : 'No info'}
			/>

			<RecipeList
				label="Ingredients"
				detailsList={
					ingredientLines && ingredientLines.length > 0
						? createList(ingredientLines)
						: 'No info'
				}
			/>

			<RecipeLine
				label="Total time"
				recipeDetail={totalTime ? totalTime + 'min' : 'No info'}
			/>
		</RecipeSummaryContainer>
	);
};

export default RecipeSummary;
