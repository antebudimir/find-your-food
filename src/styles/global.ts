import { createGlobalStyle } from 'styled-components';
import RalewayRegular from 'fonts/Raleway-Regular.woff';
import RalewayBold from 'fonts/Raleway-Bold.woff';
import RalewayExtraBold from 'fonts/Raleway-ExtraBold.woff';

export const GlobalStyles = createGlobalStyle`
@font-face {
	font-family: 'Raleway Regular';
	src: url(${RalewayRegular}) format('woff');
	font-weight: normal;
	font-style: normal;
	font-display: swap;
}

@font-face {
	font-family: 'Raleway Bold';
	src: url(${RalewayBold}) format('woff');
	font-weight: bold;
	font-style: normal;
	font-display: swap;
}


@font-face {
    font-family: 'Raleway Extra Bold';
    src: url(${RalewayExtraBold}) format('woff');
    font-weight: bold;
    font-style: normal;
    font-display: swap;
}

* {
	outline: none;
	box-sizing: border-box;
	margin: 0;
	padding: 0;
	scroll-behavior: smooth;
}

html {
    font-family: 'Raleway Regular';
	word-wrap: break-word;
}

button,
input {
	border: none;
	display: block;
}

input,
select,
textarea,
button {
	color: inherit;
	font-size: inherit;
}

input,
select,
textarea {
	font-family: inherit;
}

h1 {
    font-family: 'Raleway Extra Bold';
}

h2,
button,
a {
	font-family: 'Raleway Bold';
}

ul,
li {
	list-style: none;
}

a {
	text-decoration: none;
}

img {
	display: block;
	max-width: 100%;
}
`;
