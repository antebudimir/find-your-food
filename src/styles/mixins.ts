import { css } from 'styled-components';

interface FlexLayoutProps {
	direction?: string;
	wrap?: string;
}

interface FlexAlignmentProps {
	justify?: string;
	align?: string;
}

export const FlexLayout = ({ direction, wrap }: FlexLayoutProps) => css`
	display: flex;
	flex-flow: ${direction} ${wrap};
`;

export const FlexAlignment = ({ justify, align }: FlexAlignmentProps) => css`
	justify-content: ${justify};
	align-items: ${align};
`;

export const hiddenVisually = css`
	position: absolute;
	clip: rect(0 0 0 0);
	clip-path: inset(50%);
	overflow: hidden;
	width: 1px;
	height: 1px;
	white-space: nowrap;
`;

export const sticky = css`
	display: flex;
	flex-flow: column nowrap;
	min-height: 89.6vh;
`;
