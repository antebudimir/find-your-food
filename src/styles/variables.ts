export const variables = {
	borderRadius: {
		primary: '0.125rem',
	},
	outline: {
		primary: '1px',
	},
	shadow: {
		primary: '#3b3f20 0px 2px 5px 0px',
		secondary: '#d45464 0px 2px 5px 0px',
		tertiary: '5px 5px 20px rgba(0, 0, 0, 0.7)',
		quaternary: '5px 5px 5px rgba(0, 0, 0, 0.7)',
	},
	sizing: {
		inputDesktopWidth: '28rem',
		summaryLinkWidth: '18.9rem', // coresponds to the width of the image
		summaryContainerWidth: '28rem',
		mainContainerWidth: '60rem',
	},
	spacing: {
		primary: '1rem',
		secondary: '0.5rem 1rem',
		tertiary: '2rem',
		quaternary: '1.5rem',
	},
	color: {
		primary: '#d45464',
		primaryLighter: '#ff6478',
		secondary: ' #ffffff',
		tertiary: '#3b3f20',
	},
	fontSize: {
		primary: '1rem',
		secondary: '1.3rem',
		tertiary: '1.5rem',
		quaternary: '2rem',
		quinary: '120%',
	},
	timing: {
		primary: '200ms',
	},
	lineHeight: {
		primary: '1.5',
		secondary: '2',
	},
};
