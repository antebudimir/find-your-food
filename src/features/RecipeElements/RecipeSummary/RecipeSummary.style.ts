import styled from 'styled-components';
import { FlexLayout, FlexAlignment } from 'styles/mixins';
import Image from 'components/Elements/Image/Image';
import { Link } from 'react-router-dom';

export const RecipeSummaryContainer = styled.section`
	${FlexLayout({ direction: 'column' })};
	${FlexAlignment({ align: 'center' })};
	box-shadow: ${(props) => props.theme.shadow.primary};
	padding: ${(props) => props.theme.spacing.primary};
	width: 100%;
	max-width: ${(props) => props.theme.sizing.summaryContainerWidth};

	:not(:first-child):not(:last-child) {
		margin-block: ${(props) => props.theme.spacing.primary};
	}

	@media screen and (min-width: 700px) {
		:not(:first-child):not(:last-child) {
			margin-block: initial;
		}
	}
`;

export const SummaryTitle = styled.h2`
	margin-block-end: ${(props) => props.theme.spacing.primary};
	color: ${(props) => props.theme.color.tertiary};
`;

export const SummaryImage = styled(Image)`
	border-radius: ${(props) => props.theme.borderRadius.primary};
	box-shadow: ${(props) => props.theme.shadow.primary};
`;

export const SummaryLink = styled(Link)`
	box-shadow: ${(props) => props.theme.shadow.primary};
	width: 100%;
	max-width: ${(props) => props.theme.sizing.summaryLinkWidth};
	border-radius: ${(props) => props.theme.borderRadius.primary};
	margin-block-start: ${(props) => props.theme.spacing.primary};
	padding: ${(props) => props.theme.spacing.secondary};
	background-color: ${(props) => props.theme.color.primary};
	color: ${(props) => props.theme.color.secondary};

	&:hover,
	&:focus {
		background-color: ${(props) => props.theme.color.primaryLighter};
		transition: background-color ${(props) => props.theme.timing.primary};
	}

	@media screen and (min-width: 700px) {
		font-size: ${(props) => props.theme.fontSize.secondary};
	}
`;
