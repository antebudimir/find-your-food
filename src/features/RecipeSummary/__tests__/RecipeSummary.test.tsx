import { ThemeProvider } from 'styled-components';
import { variables } from 'styles/variables';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';
import RecipeSummary from '../RecipeSummary';

describe('<RecipeSummary /> component', () => {
	test('if the Component matches snapshot', () => {
		const component = TestRenderer.create(
			<ThemeProvider theme={variables}>
				<RecipeSummary />
			</ThemeProvider>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
