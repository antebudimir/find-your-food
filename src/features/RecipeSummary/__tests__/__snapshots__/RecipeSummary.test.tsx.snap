// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeSummary /> component if the Component matches snapshot 1`] = `
<section
  className="sc-eCYdqJ hSdRLg"
  data-testid="summary-id"
>
  <h2
    className="sc-jSMfEi eAYKRK"
  >
    Title not available
  </h2>
  <img
    alt="Image description not available"
    className="sc-gKXOVf jQmnMb"
    src="Photo not available"
  />
  <p
    className="sc-bczRLJ hHaKtq"
    data-testid="recipe-line"
  >
    <span
      className="sc-gsnTZi iqEmM"
    >
      Meal type
      :
    </span>
     
    No info
  </p>
  <p
    className="sc-bczRLJ hHaKtq"
    data-testid="recipe-line"
  >
    <span
      className="sc-gsnTZi iqEmM"
    >
      Meal type
      :
    </span>
     
    No info
  </p>
  <ul
    className="sc-dkzDqf gyuaL"
  >
    <span
      className="sc-hKMtZM YTvug"
    >
      Ingredients
      :
    </span>
     
    No info
  </ul>
  <p
    className="sc-bczRLJ hHaKtq"
    data-testid="recipe-line"
  >
    <span
      className="sc-gsnTZi iqEmM"
    >
      Total time
      :
    </span>
     
    No info
  </p>
</section>
`;
